### PR TITLE
WIP: feat: add minio cache

### LINKED ISSUE #464: [Feature]: Add minio remote cache support in the operator
### Checklist

- [X] I've searched for similar issues and couldn't find anything matching
- [X] I've discussed this feature request in the [K8sGPT Slack](https://k8sgpt.slack.com) and got positive feedback

### Is this feature request related to a problem?

None

### Problem Description

_No response_

### Solution Description

We have recently added Minio as another [remote cache](https://github.com/k8sgpt-ai/k8sgpt/blob/main/README.md?plain=1#L444) storage, to hold K8sGPT's findings.

In order to have feature parity, it will be great to add it in the operator's config as well

### Benefits

Feature parity with K8sGPT

### Potential Drawbacks

_No response_

### Additional Information

There is already a GRPC proto [schema](https://github.com/k8sgpt-ai/schemas/blob/main/protobuf/schema/v1/server-service.proto#L88) to support minio S3-like api

and the original [PR](https://github.com/k8sgpt-ai/k8sgpt/pull/1048) has included the server mode changes

### PR DESCRIPTION
**Pull request by aghilish**
Closes #464 
This pull request enables using minio backend as s3 cache alternative. 

## ✅ Checks

- [x] My pull request adheres to the code style of this project
- [x] My code requires changes to the documentation
- [x] I have updated the documentation as required
- [x] All the tests have passed

### CODE REVIEW

**Comment by aghilish** [comment]
@arbreezy Hi, 

As an edge case when I use bedrock as the AI provider, client config can not pick up the minio access key and credentials from the cache secret and keeps using the aws_access_key_id and aws_secret_access_key.

please see `Minio` section in the README to quickly reproduce the issue.

**Comment by arbreezy** [comment]
Are those envs not deprecated now?
https://github.com/bitnami/charts/issues/6726

File: pkg/resources/k8sgpt.go [hunk 1]
```diff
@@ -266,8 +266,15 @@ func GetDeployment(config v1alpha1.K8sGPT, outOfClusterMode bool, c client.Clien
 			addRemoteCacheEnvVar("AZURE_TENANT_ID", "azure_tenant_id")
 			addRemoteCacheEnvVar("AZURE_CLIENT_SECRET", "azure_client_secret")
 		} else if config.Spec.RemoteCache.S3 != nil {
-			addRemoteCacheEnvVar("AWS_ACCESS_KEY_ID", "aws_access_key_id")
```
Comment (arbreezy) [inline pkg/resources/k8sgpt.go:269]: Why do we need different env variables although we are using the same aws client,  in theory this shouldn't matter if it's s3 native or minio?

**Comment by arbreezy** [comment]
> @arbreezy Hi,
> 
> As an edge case when I use bedrock as the AI provider, client config can not pick up the minio access key and credentials from the cache secret and keeps using the aws_access_key_id and aws_secret_access_key.
> 
> please see `Minio` section in the README to quickly reproduce the issue.

Good shout, 

this happens cause we override the same env variable with the bedrock's secret key which is added later thus overwriting the cache's secret. 

I think the options are a) we initialise a minio native client b) we use profiles with the latter we need to create files IIRC

**Comment by arbreezy** [comment]
I think we should adress the AWS env variables in another PR. The issue exists with s3 as remote cache + bedrock. 

@aghilish you will have to add the updated CRD in helm templates too

**Review by arbreezy — CHANGES_REQUESTED** [review]
